python/pyspark/sql/readwriter.pyi

```diff
@@ -125,9 +131,9 @@ class DataFrameReader(OptionUtils):
         url: str,
         table: str,
         column: str,
-        lowerBound: int,
-        upperBound: int,
-        numPartitions: int,
+        lowerBound: Union[int, str],
+        upperBound: Union[int, str],
+        numPartitions: Union[int, str],
```

Comment:
I'll just stick this to int. BTW for `lowerBound` and `upperBound`, now they can accept timestamp and dates in string formats, @zero323